ouroboros-consensus/src/Ouroboros/Consensus/Mempool/Impl.hs

```diff
@@ -62,8 +67,11 @@ openMempool registry chainDB cfg = do
 
 -- | Internal state in the mempool
 data InternalState blk = IS {
-     -- | Transactions currently in the mempool
-      isTxs :: TxSeq (GenTx blk)
+      -- | Transactions currently in the mempool
+      isTxs :: TxSeq (GenTxId blk, GenTx blk)
+
+      -- |
+    , isTxMap :: Map (GenTxId blk) (GenTx blk)
```

Comment:
Do we need this for anything?

Reply:
Yes, a `lookup` is done on this `Map` in the implementation for `getTxById`. Keeping this `Map` around could be avoided if we're willing to convert `isTxs` (a `TxSeq`) to a `Map` every time `getTxById` is called. However, this doesn't seem worth it to me.

Reply:
In reference to https://github.com/input-output-hk/ouroboros-network/pull/678/commits/10a0d395672cab7ebef6b0e65591a58722ebe4ca#r298587558

Going to drop `isTxMap`. Like you said, it doesn't seem that we have any need for `getTxById`.